synapse/handlers/sync.py

```diff
@@ -1817,11 +1817,34 @@ async def _generate_sync_entry_for_rooms(
             )
             sync_result_builder.now_token = now_token
 
+        # Retrieve rooms that got un partial stated in the meantime, only useful in case
+        # of a non lazy-loading-members sync.
+        un_partial_stated_rooms = set()
+        if not sync_result_builder.sync_config.filter_collection.lazy_load_members():
+            un_partial_state_rooms_since = 0
+            if sync_result_builder.since_token is not None:
+                un_partial_state_rooms_since = int(
+                    sync_result_builder.since_token.un_partial_state_rooms_key
+                )
+
+            un_partial_state_rooms_now = int(
+                sync_result_builder.now_token.un_partial_state_rooms_key
+            )
+            if un_partial_state_rooms_since != un_partial_state_rooms_now:
```

Comment:
I think we already have this check in the body of `get_un_partial_stated_rooms_between`? I'd be tempted to remove it here, just to keep this function more readable.